### PR TITLE
chore(backend): fix flaky test that made wrong assumptions about order being guaranteed

### PR DESCRIPTION
**Pull request by corneliusroemer**
resolves https://github.com/loculus-project/loculus/issues/3235

Came up frequently while working on #3232 so I decided to fix it (but break out into separate PR to keep that PR clean)

### CODE REVIEW

File: backend/src/test/kotlin/org/loculus/backend/controller/submission/DeleteSequencesEndpointTest.kt [hunk 1]
```diff
@@ -317,8 +327,7 @@ class DeleteSequencesEndpointTest(
             .andExpect(status().isOk)
             .andExpect(content().contentType(MediaType.APPLICATION_JSON_VALUE))
             .andExpect(jsonPath("\$.length()").value(accessionVersions.size))
-            .andExpect(jsonPath("\$[0].accession").value(accessionVersions.first().accession))
-            .andExpect(jsonPath("\$[0].version").value(accessionVersions.first().version))
+            .andExpect(jsonPath("\$[*]", containsInAnyOrder(*expectedPairs.toTypedArray())))
```
Comment (fengelniederhammer) [inline backend/src/test/kotlin/org/loculus/backend/controller/submission/DeleteSequencesEndpointTest.kt:330]:
```suggestion
            .andExpect(jsonPath("\$[*]", containsInAnyOrder(expectedPairs)))
```

**Review by fengelniederhammer — APPROVED** [review]
Nice :)

File: backend/src/test/kotlin/org/loculus/backend/controller/submission/DeleteSequencesEndpointTest.kt (hunk 1 above)
Comment (corneliusroemer) [reply backend/src/test/kotlin/org/loculus/backend/controller/submission/DeleteSequencesEndpointTest.kt:330]: Have you tested it? I'll commit it just for fun as revert is easy

Comment (corneliusroemer) [reply backend/src/test/kotlin/org/loculus/backend/controller/submission/DeleteSequencesEndpointTest.kt]: nice it worked!

**Pull request merged**